adadamp/_dist.py

```diff
@@ -71,50 +95,79 @@ def gradient(
     where `l` is the loss function for a single example.
 
     """
-    model, _ = model_opt
+    
+    # Workaround: Gradients should be cleared when entering this funciton, 
+    #     but at this moment this behavior is not occuring
+    old_model, _ = model_opt
+    model = deepcopy(old_model)
+    
+    if get_model_grads(model) != 0:
+        assert False, "ERROR Gradients not zero'd at grad time"
+        print("NOTE: Gradients not zero'd at grad time")
+            
+    # print(id(model)) 
+    
     start = time()
+    
+    # set up data
     data_target = [train_set[i] for i in idx]
     _inputs = [d[0].reshape(-1, *d[0].size()) for d in data_target]
     _targets = [d[1] for d in data_target]
-    inputs = torch.cat(_inputs)
-    targets = torch.tensor(_targets)
+    inputs = torch.cat(_inputs).to(device)
+    targets = torch.tensor(_targets).to(device)
+    
+    # split by max bastch size    
+    Data = torch.split(inputs, max_bs)
+    Target = torch.split(targets, max_bs)
+    
+    # run through in batches tracking net result
+    loss_agg = 0
+    n_items = 0
+    for data, target in zip(Data, Target):
+        data, target = data.to(device), target.to(device)
+        outputs = model(data)
+        _loss = loss(outputs, target)
+        _loss.backward()
+        loss_agg += _loss.item()
+        n_items += len(outputs)
 
     assert model.training, "Model should be in training model"
-    outputs = model(inputs)
-
-    _loss = loss(outputs, targets)
-    _loss.backward()
+    # assert get_model_grads(model) == 0, "Gradients not cleared before loss.backward()"
+    
+    
     grads: Dict[str, torch.Tensor] = {k: v.grad for k, v in model.named_parameters()}
     elapsed = time() - start
     return {
-        "_num_data": len(outputs),
+        "_num_data": n_items,
         "_time": elapsed,
-        "_loss": float(_loss.item()),
+        "_loss": loss_agg / n_items,
```

Comment:
```suggestion
        "_loss": float(loss_agg.item()),
```

The `_loss` key is never examined, and unnecessary because of the following line in `_update_model`: https://github.com/stsievert/adadamp/blob/9bccb9d2cd9bbf867de0b413046b4b787bc0d6ea/adadamp/_dist.py#L160

Reply:
Okay. How would not averaging the loss impact the model training then? When we saw different results from the paper, weren't we assuming it was the result of simply summing all of the losses?

Reply:
I misunderstood what "summing the losses" meant. Ask me when we meet.